### PR TITLE
[CHG] Save as template

### PR DESCRIPTION
**Pull request by andreparames**
Implements essentially the same feature as in Odoo's mail composer, by creating an email.template from the subject/body and then linking the current message to it. 

I've created a new file for adding the code using the new API. If I should port the original file instead, just let me know.

refs #28189

### CODE REVIEW

File: odoo_addons/mozaik_communication/wizard/distribution_list_mass_function_v8.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class DistributionListMassFunction(models.TransientModel):
+    _inherit = 'distribution.list.mass.function'
+
+    @api.multi
+    def save_as_template(self):
+        template_name = u"Mass Function: {subject}"
+        model = self.env['ir.model'].search(
+            [('model', '=', self._name)], limit=1)
```
Comment (Olivier-LAURENT) [inline odoo_addons/mozaik_communication/wizard/distribution_list_mass_function_v8.py]: @andreparames 
Here you have to force the model to be `email.coordinate`, not the wizard model.

File: odoo_addons/mozaik_communication/wizard/distribution_list_mass_function_v8.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#     be useful but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the
+#     GNU Affero General Public License
+#     along with mozaik_communication.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (Olivier-LAURENT) [inline odoo_addons/mozaik_communication/wizard/distribution_list_mass_function_v8.py]: @andreparames 
cosmetic: you can now use the new short style of comment (as generated by mr.bob).

File: odoo_addons/mozaik_communication/wizard/distribution_list_mass_function_v8.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+from openerp import models, api
```
Comment (Olivier-LAURENT) [inline odoo_addons/mozaik_communication/wizard/distribution_list_mass_function_v8.py]: @andreparames 
cosmetic also: usually `... api, models...`

File: odoo_addons/mozaik_communication/tests/test_mass_function.py [hunk 4]
```diff
@@ -36,3 +36,19 @@ def test_onchange_subject_and_template(self):
...
+            'e_mass_function': 'email_coordinate_id',
+            'distribution_list_id': evr_lst_id,
+            'subject': 'TEST1',
+            'body': '<p>hello</p>',
+        }
+        wizard = mfct_obj.with_context({'in_mozaik_user': True}).create(vals)
+        wizard.save_as_template()
+        self.assertTrue(bool(wizard.email_template_id))
+        self.assertEqual(wizard.email_template_id.subject, vals['subject'])
+        self.assertEqual(wizard.email_template_id.body_html, vals['body'])
```
Comment (Olivier-LAURENT) [inline odoo_addons/mozaik_communication/tests/test_mass_function.py:54]: @andreparames 
Check also here for model. It must be `email.coordinate`. Thx.

**Review by Olivier-LAURENT — COMMENTED** [review]
Just one important thing IMO. Thx

**Comment by andreparames** [comment]
Thanks for the review! I've addressed those points.

**Review by Olivier-LAURENT — APPROVED** [review]
Thanks for the PR

**Pull request merged**